### DISCUSSION TITLE
Flows - Optionally send flow accountability with webhook/request

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#17577] [Feature Requests]
### Summary

The request operation in flows is a super useful feature for a ton of use cases.

One of the use cases we run into a lot is that we want to hit a custom endpoint on a cron schedule. The pattern we implement is the ability to kick off a process from a custom endpoint in our application UI manually, and also to have these processes run on a schedule.

The endpoints are secure and controlled by the accountability of the user that executed them - in many cases limited to admin or manager roles.

It would be convenient if the request operation could optionally send along the accountability JWT of the user the flow is executing under along with the request. This would prevent us from needing to hard-code static auth tokens in header params when executing a scheduled request operation.

I'm not sure how much of an edge-case this is, or how many other people run into the issue. If there's interest in this feature, I don't mind preparing a PR.

### Motivation

The motivation is to simplify the use case where flows are used to call back to an authenticated directus endpoint, to have the endpoint execute under the permissions of the user executing the flow.

### Detailed Design

In https://github.com/directus/directus/blob/main/api/src/operations/request/index.ts the accountability would need to be passed in the same way item-create, etc... works.

The auth JWT would need to be created from the accountability and optionally added to the Axios headers of the request.

### Must Have:
- The ability to pass the accountability JWT along with a request operation in directus flows

### Should Have:
- A UI enhancement to the Operation options to optionally include the flow accountability with the request

### Drawbacks

Even though implementation seems pretty simple, it may be such a rare use case that there's not much interest in the feature and not worth the time.

### Alternatives

Statically passing auth tokens in request headers and parsing/using them in a custom endpoint. Static auth tokens work for this, but introduce a bit of security nail-biting.

Another approach is to add the directus sdk or axios/fetch/etc... to [FLOWS_EXEC_ALLOWED_MODULES environment variable](https://docs.directus.io/self-hosted/config-options.html#security) and to use a script operation to authenticate the directus sdk instance and issue requests.

### Adoption Strategy

This shouldn't affect existing users, it's just an enhancement to existing functionality.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
